src/client/components/vision2/camera/world_line.ts

```diff
@@ -0,0 +1,137 @@
+import * as THREE from 'three'
+
+import { Vector2 } from '../../../math/vector2'
+import { Vector3 } from '../../../math/vector3'
+import { Vector4 } from '../../../math/vector4'
+import { mesh } from '../../three/builders'
+import { planeGeometry } from '../../three/builders'
+import { rawShader } from '../../three/builders'
+import { shaderMaterial } from '../../three/builders'
+import { createUpdatableComputed } from '../../three/create_updatable_computed'
+import { Canvas } from '../../three/three'
+
+import { Lens } from './model'
+import fragmentShader from './shaders/world_line.frag'
+import vertexShader from './shaders/world_line.vert'
+
+export interface ConeSegment {
+  /** The normal axis of the plane in camera space. Only needed if start and end are parallel or anti-parallel. */
+  axis?: Vector3,
+  /** The camera space vector pointing to the start of the line segment. */
+  start: Vector3,
+  /** The camera space kvector pointing to the end of the line segment. */
+  end: Vector3,
+  /** The colour of the line to draw. */
+  color?: Vector4,
+  /** The width of the line to draw on the screen in pixels. */
+  lineWidth?: number
+}
+
+export class WorldLine {
+  constructor(
+    private readonly canvas: Canvas,
+    private readonly lens: Lens,
+  ) {
+  }
+
+  static of(canvas: Canvas, lens: Lens) {
+    return new WorldLine(canvas, lens)
+  }
+
+  /** Draws a plane projected to infinity in world space. */
+  plane({ axis, color, lineWidth }: { axis: Vector3, color?: Vector4, lineWidth?: number }) {
+    // Pick an arbitrary orthogonal vector
+    const start = Vector3.fromThree(
+      (!axis.x && !axis.y)
+        ? new THREE.Vector3(0, 1, 0)
+        : new THREE.Vector3(-axis.y, axis.x, 0).normalize(),
+    )
+    return this.coneSegment({ axis, start, end: start, color, lineWidth })
+  }
+
+  /** Draws a segment of a plane projected to infinity in world space. */
+  planeSegment(segment: ConeSegment) {
```

Comment:
I think it's incorrect to make this take a ConeSegment.
In a ConeSegment the axis variable should not be optional but here it is.